test/events/impression.spec.ts

```diff
@@ -0,0 +1,219 @@
+
+describe('impression tests', () => {
+    it('should accept valid impression event', (done) => {
+        const expectedEvent = {
+            impression: {
+                impressionType: 'recommendation',
+                product: {
+                    productId: 'asdfasd',
+                    category: 'boats',
+                    collection: 'boatssrus',
+                    title: 'boats',
+                    sku: 'asdfasf98',
+                    price: 100.21,
+                },
+            },
+            eventType: 'impression',
+            customer: {
+                id: 'testcustomer',
+                area: 'area',
+            },
+            visit: {
+                customerData: {
+                    visitorId: 'visitor',
+                    sessionId: 'session',
+                },
+                generated: {
+                    uri: '',
+                    timezoneOffset: 240,
+                    localTime: '2016-08-14T14:05:26.872Z',
+                },
+            },
+        };
+
+        const gbTrackerCore = new GbTracker(expectedEvent.customer.id, expectedEvent.customer.area);
+
+        gbTrackerCore.setInvalidEventCallback(() => {
+            done('fail');
+        });
+
+        gbTrackerCore.__getInternals().sendEvent = (event: any) => {
+
+            expect(event.clientVersion.raw).to.not.be.undefined;
+            expect(event.impression.impressionType).to.eql(expectedEvent.impression.impressionType);
+            expect(event.impression.product).to.eql(expectedEvent.impression.product);
+            expect(event.eventType).to.eql(expectedEvent.eventType);
+            expect(event.customer).to.eql(expectedEvent.customer);
+            expect(event.visit.customerData).to.eql(expectedEvent.visit.customerData);
+            expect(event.visit.generated.timezoneOffset).to.not.be.undefined;
+            expect(event.visit.generated.localTime).to.not.be.undefined;
+            done();
+        };
+
+        gbTrackerCore.setVisitor(expectedEvent.visit.customerData.visitorId, expectedEvent.visit.customerData.sessionId);
+
+        gbTrackerCore.sendImpressionEvent({
+            impressionType: expectedEvent.impression.impressionType,
+            product: expectedEvent.impression.product,
+        });
+    });
+
+    it('should reject invalid impression event not in right format', (done) => {
+        const gbTrackerCore = new GbTracker('testcustomer', 'area');
+
+        gbTrackerCore.__getInternals().sendEvent = (event: any) => {
+
+            done('fail');
+        };
+
+        gbTrackerCore.setVisitor('visitor', 'session');
+
+        gbTrackerCore.setInvalidEventCallback((event: any, error: any) => {
+            expect(error).to.match(/impression: is missing/);
+            done();
+        });
+        gbTrackerCore.sendImpressionEvent({
```

Comment:
schema doesn't seem to match here. Needs to use the Impression partial (which has impressionType and products inside it)